### PR TITLE
Add logging and cache for the wharf url

### PR DESCRIPTION
**Pull request by briandant**

### CODE REVIEW

**Review by tomaszzielinski — APPROVED** [review]
Looks good AFAIC but I'm not sure how the logging might work with Tahoe so that part would be for @bezidejni.

File: launchcontainer/launchcontainer.py [hunk 1]
```diff
@@ -106,26 +96,51 @@ class LaunchContainerXBlock(XBlock):
...
-            # a SiteConfig object is changed.
+            (
+                'SiteConfig', site_wharf_url
+            ),
+            # TODO: Maybe we can cache this and set up a signal
+            # to update this value if a SiteConfig object is changed.
             # A string: the currently supported implementation.
-            settings.ENV_TOKENS.get('LAUNCHCONTAINER_WHARF_URL'),
+            (
+                "ENV_TOKENS['LAUNCHCONTAINER_WHARF_URL']",
```
Comment (bezidejni) [inline launchcontainer/launchcontainer.py]: Why are these now tuples instead of just values as before?

File: launchcontainer/launchcontainer.py [hunk 2]
```diff
@@ -106,26 +96,51 @@ class LaunchContainerXBlock(XBlock):
...
+        def is_valid(url):
+            try:
+                validator(url)
+            except validators.ValidationError:
+                return False
+            else:
+                return True
+
+        logger.debug("XBlock-launchcontainer urls attempted: {}".format(urls))
+        if IS_OPENEDX_ENVIRON:
```
Comment (bezidejni) [inline launchcontainer/launchcontainer.py]: Not sure I understand this, when would you run an XBlock that's not in an Open edX environment?

Comment (briandant) [reply launchcontainer/launchcontainer.py]: We run this locally in the `xblock-sdk` for development.

File: launchcontainer/launchcontainer.py (hunk 1 above)
Comment (briandant) [reply launchcontainer/launchcontainer.py]: The values in the site config and in the env vars are simply strings, but I changed this to a tuple to make the logs more transparent.

**Comment by briandant** [comment]
@tomaszzielinski / @bezidejni: I just added another commit that caches the wharf url value. I wanted to implement this at some point, and given that the cache was confusing during our last install to Tahoe staging, I just went ahead and did it now. Please let me know if this looks okay, or if I have missed anything. Many thanks!

File: launchcontainer/launchcontainer.py (hunk 1 above)
Comment (briandant) [reply launchcontainer/launchcontainer.py]: I do intend to slim this down in the future, such that we merely log success and failures, and still only at the `DEBUG` level.

**Comment by briandant** [comment]
@bezidejni: It just occurred to me that since the URL was probably failing due to cache issues—I saw and fixed the same behavior locally—we might not need to bother with enabling the debug logger if that's going to be a hassle for you.

File: launchcontainer/launchcontainer.py (hunk 1 above)
Comment (bezidejni) [reply launchcontainer/launchcontainer.py]: Yeah, it does complicate the code a bit if they're tuples, and the benefits aren't that big.

**Review by bezidejni — APPROVED** [review]

**Comment by briandant** [comment]
Yeah, agreed. My original idea was to have some sort of `key:value` 
thing going on, then as I parsed through them, I could log the successes 
and failures. I’m now thinking that I’ll just drop all of this soon 
and only support the `SiteConfiguration` object.

**Pull request merged**